src/Dao/EnwikiDao.php

```diff
@@ -44,26 +48,71 @@ public function __construct(
 	) {
 		parent::__construct( $dsn, $user, $pass, $logger );
 		$this->wikipedia = $wiki;
+		$this->api = \Mediawiki\Api\MediawikiApi::newFromApiEndpoint( $wiki . '/w/api.php' );
 	}
 
 	/**
-	 * Get details on multiple revisions
+	 * Get details on given revisions
 	 *
 	 * @param $diffs array Revision IDs
-	 * @return array full revision rev_id, rev_user, rev_usertext,
-	 *   user_editcount and user_name of the revision
+	 * @return array Associative array by revids containing editor and timestamp
 	 */
-	public function getRevisionDetailsMulti( $diffs ) {
-		$query = self::concat(
-			'SELECT r.rev_id, r.rev_user, r.rev_page, r.rev_user_text,
-			u.user_editcount, u.user_name, p.page_namespace',
-			'FROM revision r',
-			'LEFT JOIN user u ON r.rev_user = u.user_id',
-			'LEFT JOIN page p ON r.rev_page = p.page_id',
-			'WHERE r.rev_id IN (' . implode( ',', array_fill( 0, count( $diffs ), '?' ) ) . ')'
-		);
-		$result = $this->fetchAll( $query, $diffs );
-		return $result;
+	public function getRevisionDetails( $diffs ) {
+		// get the revisions synchronously
+		$result = $this->apiQuery( [
+			'revids' => implode( '|', $diffs ),
+			'prop' => 'revisions',
+			'rvprop' => 'user|timestamp|ids'
+		] )['query'];
+
+		$data = [];
+
+		// fill in nulls for deleted revisions
+		if ( isset( $result['badrevids'] ) ) {
+			foreach ( $result['badrevids'] as $revision ) {
+				$data[$revision['revid']] = null;
+			}
+		}
+
+		foreach ( $result['pages'] as $page ) {
+			$revisions = $page['revisions'];
+
+			if ( isset( $revisions ) ) {
+				foreach ( $revisions as $revision ) {
+					$data[$revision['revid']] = [
+						'revid' => $revision['revid'],
```

Comment:
It's a bit odd having the revision ID in the data twice. Do we need both of them?